external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriteTask.scala

```diff
@@ -53,7 +52,7 @@ private[kafka010] class KafkaWriteTask(
     if (producer != null) {
```

Comment:
Even it is defined as `lazy`, here it will be initialized if `producer` is not referenced (and initialized) in `send` - so if close() is guaranteed to be called, it doesn't seem to be noticeable difference.